core/services/keeper/models.go

```diff
@@ -0,0 +1,36 @@
+package keeper
+
+import "github.com/smartcontractkit/chainlink/core/store/models"
+
+type Registry struct {
+	ID                int32 `gorm:"primary_key"`
+	BlockCountPerTurn int32
+	CheckGas          int32
+	ContractAddress   models.EIP55Address
+	FromAddress       models.EIP55Address
+	JobID             int32
+	KeeperIndex       int32
+	NumKeepers        int32
+}
+
+func NewRegistry(address models.EIP55Address, from models.EIP55Address, jobID int32) Registry {
+	return Registry{
+		ContractAddress: address,
+		FromAddress:     from,
+		JobID:           jobID,
+	}
+}
+
+func (Registry) TableName() string {
+	return "keeper_registries"
+}
+
+type UpkeepRegistration struct {
+	ID                  int32 `gorm:"primary_key"`
+	CheckData           []byte
+	ExecuteGas          int32
+	RegistryID          int32
+	Registry            Registry `gorm:"association_autoupdate:false"`
```

Comment:
note in gorm v2 there is no `association_autoupdate:false` tags. By default it will create the association record if it doesn't exist. Do we want that? If not we've been using readonly tags i.e. `->` [https://gorm.io/docs/models.html#Field-Level-Permission] but in general have been heavily preferring to use raw sql